runtime/vm/ContinuationHelpers.cpp

```diff
@@ -977,6 +977,17 @@ preparePinnedVirtualThreadForUnmount(J9VMThread *currentThread, j9object_t syncO
 				syncObjectMonitor = monitorTableAt(currentThread, syncObj);
 				monitor = syncObjectMonitor->monitor;
 
+				if (J9_EVENT_IS_HOOKED(vm->hookInterface, J9HOOK_VM_MONITOR_CONTENDED_ENTERED)
+					&& J9_ARE_NO_BITS_SET(currentThread->currentContinuation->runtimeFlags, J9VM_CONTINUATION_RUNTIMEFLAG_JVMTI_CONTENDED_MONITOR_ENTER_RECORDED)
+				) {
+					/* It could be too late in yieldPinnedContinuation() that we cannot get the owner. Get the owner here. */
+					if ((void*)1 == monitor->owner) {
+						currentThread->currentContinuation->previousOwner = NULL;
+					} else {
+						currentThread->currentContinuation->previousOwner = getVMThreadFromOMRThread(vm, ((J9ThreadMonitor *)monitor)->owner);
```

Comment:
I feel the check `monitor->owner` for `0x1` should be done in `getVMThreadFromOMRThread` to ensure other use of this API would have same protection and result.

Reply:
This can be done in a separate PR. I slightly prefer adding check of `0x1` in the caller of `getVMThreadFromOMRThread()`. The name of the method "getVMThreadFromOMRThread" implies the passed in para should be an `OMRThread`. If it is `0x1`, it should not call into `getVMThreadFromOMRThread()`.